exercises/collatz-conjecture/example.exs

```diff
@@ -0,0 +1,25 @@
+defmodule CollatzConjecture do
+
+  @doc """
+  calc/1 takes number (> 1), and returns the number of steps required to get to 1 when
+  following the rules: if n is odd, multiply with 3 and add 1. if n is even, divide by 2
+  """
+  @spec calc(number :: integer) :: pos_integer
+  def calc(number) when number < 1 do
+    raise ArgumentError, "Only positive numbers are allowed"
+  end
+
+  def calc(number) do
```

Comment:
If the guard for integer is added to the second function definition, then `calc("foo")` will raise a no-clause error. If we add the guard to the first definition, we'll get the error saying only positive numbers are allowed.

The first function would the look like:
```
def calc(number) when number < 1 or not is_integer(number) do
  raise ArgumentError, "Only positive numbers are allowed"
end
```